commitrmnocb/outcome.go

```diff
@@ -108,11 +125,23 @@ func buildReport(
 		outcomeType = ReportEmpty
 	}
 
+	var lastPricesUpdate time.Time
+	tokenPrices := consensusObservation.TokenPricesArray()
+
+	// If we're updating all token prices, set the last prices update to the observed timestamp,
+	// otherwise keep it as the last update time.
+	if len(offchainCfg.TokenInfo) == len(tokenPrices) {
```

Comment:
At this step we already filtered what tokens we want to update on the registry, not everytime we'll update all token prices, this should happen when the time period to update all has passed. Otherwise it will only update a subset of tokens (which deviated from the threshold)